hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/HashMapStreamSerializer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The {@link Map} serializer
+ */
+public class HashMapStreamSerializer extends AbstractMapStreamSerializer {
+    @Override
+    public int getTypeId() {
+        return SerializationConstants.JAVA_DEFAULT_TYPE_HASH_MAP;
+    }
+
+    @Override
+    protected Map createMap(int size) {
+        return new HashMap(size);
```

Comment:
Minor: use `MapUtil#createHashMap` as this constructor is not going to size the internal table adequately because of load factor.